src/main.js

```diff
@@ -1,355 +1,30 @@
-    });
-    console.info("Monitoring feed updates", [feed]);
-    return;
-  }
-
-  feed = document.getElementById("pagelet_group_");
-  if (feed !== null) {
-    // if the user change page to https://www.facebook.com/groups/*
-    feed.querySelectorAll('div[id^="mall_post_"]').forEach(hideIfSponsored);
-    feedObserver = new MutationObserver((mutations) => {
-      mutations.forEach((mutation) => {
-        mutation.target
-          .querySelectorAll('div[id^="mall_post_"]')
-          .forEach(hideIfSponsored);
-      });
-    });
-    feedObserver.observe(feed, {
-      childList: true,
-      subtree: true,
-    });
-    console.info("Monitoring feed updates", [feed]);
-    return;
-  }
-
-  // FB5 design
-  // there's a feed div that we don't monitor yet
-  feed = document.querySelector("div[role=feed]:not([data-adblock-monitored])");
-  if (feed !== null) {
-    setFB5FeedObserver();
-    return;
-  }
-  // there's a feed loading placeholder
-  feed = document.getElementById("suspended-feed");
-  if (feed !== null) {
-    setFB5FeedObserver();
-    return;
-  }
-  // No new feed was detected
-  // Cleanup observer when there's no feed monitored left in DOM
-  if (
-    feedObserver !== null &&
-    document.querySelector("div[role=feed][data-adblock-monitored]") === null
-  ) {
-    feedObserver.disconnect();
-  }
-}
-
-const fbObserver = new MutationObserver(onPageChange);
-// wait for and observe FB5 page element
-function setupFB5PageObserver() {
-  // We are expecting to find a page div
-  const pageDiv = document.querySelector(
-    "div[data-pagelet=root] div[data-pagelet=page]"
-  );
-  // make sure there's a page element
-  if (pageDiv !== null) {
-    // trigger first page initiation
-    onPageChange();
-
-    // Facebook uses ajax to load new content so
-    // we need to observe the container of the page
-    // for any page changes
-    fbObserver.observe(pageDiv.parentNode, {
-      childList: true,
-    });
-    console.info("Monitoring page changes", [pageDiv]);
-  } else {
-    // no page div was available yet in DOM. will check again
-    setTimeout(setupFB5PageObserver, 1000);
-  }
-}
-
-let fbContent = document.getElementsByClassName("fb_content")[0];
-if (fbContent !== undefined) {
+if (isClassicFacebook()) {
   // Old Facebook design
-  // remove on first load
-  onPageChange();
-
-  // Facebook uses ajax to load new content so
-  // we need this to watch for page change
-  fbObserver.observe(fbContent, {
-    childList: true,
-  });
-  console.info("Monitoring page changes", [fbContent]);
-} else if (document.getElementById("mount_0_0") !== null) {
+  setupClassicPageObserver();
+} else if (isFB5()) {
   // if it's FB5 design
   setupFB5PageObserver();
+} else {
+  console.warn("Page element not found!");
 }
-// if we can't find ".fb_content", then it must be a mobile website.
+// if we detect a page element, then it must be a mobile website.
 // in that case, we don't need javascript to block ads
```

Comment:
I think this comment is redundant and could be omitted.

Reply:
Got it